LSPS5/README.md

```diff
@@ -0,0 +1,623 @@
+the above webhook notification object.
+
+[<LSPS0 datetime>]: ../LSPS0/common-schemas.md#link-lsps0datetime
+
+For example, if the timestamp is `2023-05-04T10:52:58.395Z`
+and the webhhok notification object is the exact ASCII sequence
+`{"jsonrpc":"2.0","method":"lsps5.goodbye","params":{}}`,
+then the JSON string to be signed is:
+
+    "LSPS5: DO NOT SIGN THIS MESSAGE MANUALLY: LSP: At 2023-05-04T10:52:58.395Z I notify {\"jsonrpc\":\"2.0\",\"method\":\"lsps5.goodbye\",\"params\":{}}"
+
+The message to be signed is the contained string in UTF-8
+format, without a trailing `NUL` character.
+For the above example, the message to be signed would have the
+following hex dump:
+
+```
+00000000: 4c53 5053 353a 2044 4f20 4e4f 5420 5349  LSPS5: DO NOT SI
+00000010: 474e 2054 4849 5320 4d45 5353 4147 4520  GN THIS MESSAGE
+00000020: 4d41 4e55 414c 4c59 3a20 4c53 503a 2041  MANUALLY: LSP: A
+00000030: 7420 3230 3233 2d30 352d 3034 5431 303a  t 2023-05-04T10:
+00000040: 3532 3a35 382e 3339 355a 2049 206e 6f74  52:58.395Z I not
+00000050: 6966 7920 7b22 6a73 6f6e 7270 6322 3a22  ify {"jsonrpc":"
+00000060: 322e 3022 2c22 6d65 7468 6f64 223a 226c  2.0","method":"l
+00000070: 7370 7335 2e67 6f6f 6462 7965 222c 2270  sps5.goodbye","p
+00000080: 6172 616d 7322 3a7b 7d7d                 arams":{}}
+```
+
+The resulting signature is a zbase32-formatted string.
+
+The timestamp, signature, and webhook notification object are sent
+to the notification delivery service based on the protocol of the
+webhook:
+
+* For HTTPS webhooks, the LSP makes a `POST` request, with
+  the webhook notification object as the exact `POST` request
+  body, and with additional HTTP headers below:
+  * `x-lsps5-timestamp` - the LSP-side timestamp, in ISO8601
+     date format `YYYY-MM-DDThh:mm:ss.uuuZ`, for example
+    `2023-05-04T10:14:23.853Z`.
+  * `x-lsps5-signature` - the LSP-generated signature, as a
+    zbase32 string.
+
+The notification delivery service, on being contacted via a
+registered webhook:
+
+* For HTTPS webhooks:
+  * MUST check that the webhook was contacted via a
+    `POST` HTTPS request.
+  * MUST validate that the URI (hostname, path, and/or
+    `GET` parameters) contains:
+    * The LSP node ID.
+    * A valid authorization for the LSP to wake the client.
+    * A way to identify the target device(s) the client is
+      running on.
+    * Or some token that it issued on registration of the
+      above information.
+  * MUST check that the `x-api-timestamp` and
+    `x-api-signature` HTTP headers exist and are the
+    expected formats.
+  * MUST validate that the `POST` request body is a valid
+    [JSON-RPC 2.0 Notification Object][], which is the
+    webhook notification object.
+
+The notification delivery service, once it has received
+the timestamp, signature, and webhook notification object:
+
+* MUST check that the timestamp is within 10 minutes
+  (before or after) of its local time.
```

Comment:
@Jasonvdb Do you know how accurate mobile phone clocks are? Is it reasonable to expect the phone clock to be in sync with the LSP clock within a 10min time window?